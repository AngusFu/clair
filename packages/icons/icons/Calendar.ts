export const Calendar = `
<svg viewBox="0 0 28 28">
  <g fill="none" fill-rule="evenodd">
    <path d="M1 5v20a2 2 0 002 2h22a2 2 0 002-2V5a2 2 0 00-2-2H3a2 2 0 00-2 2z" stroke="currentColor"/>
    <path d="M7 1v4M21 1v4" stroke="currentColor" stroke-linecap="round"/>
    <path d="M1 9h26H1z" stroke="currentColor"/>
    <path d="M7 21h2M7 15h2M13 21h2M13 15h2M19 21h2M19 15h2" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round"/>
  </g>
</svg>`;
