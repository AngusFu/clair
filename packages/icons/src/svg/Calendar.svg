<?xml version="1.0" encoding="UTF-8"?>
<svg width="28px" height="28px" viewBox="0 0 28 28" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 56.2 (81672) - https://sketch.com -->
    <title>Calendar</title>
    <desc>Created with Sketch.</desc>
    <g id="Calendar" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g>
            <rect id="蒙版" opacity="0.200000003" x="0" y="0" width="28" height="28"></rect>
            <path d="M1,5 L1,25 C1,26.1045695 1.8954305,27 3,27 L25,27 C26.1045695,27 27,26.1045695 27,25 L27,5 C27,3.8954305 26.1045695,3 25,3 L3,3 C1.8954305,3 1,3.8954305 1,5 Z" id="路径-2" stroke="#6E6E6E" stroke-width="2"></path>
            <path d="M7,1 L7,5" id="路径" stroke="#6E6E6E" stroke-width="2" stroke-linecap="round"></path>
            <path d="M21,1 L21,5" id="路径-2" stroke="#6E6E6E" stroke-width="2" stroke-linecap="round"></path>
            <path d="M1,9 L27,9 L1,9 Z" id="形状结合-path" stroke="#6E6E6E" stroke-width="2"></path>
            <path d="M7,21 L9,21" id="路径-3" stroke="#6E6E6E" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"></path>
            <path d="M7,15 L9,15" id="路径-4" stroke="#6E6E6E" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"></path>
            <path d="M13,21 L15,21" id="路径-3" stroke="#6E6E6E" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"></path>
            <path d="M13,15 L15,15" id="路径-4" stroke="#6E6E6E" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"></path>
            <path d="M19,21 L21,21" id="路径-3" stroke="#6E6E6E" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"></path>
            <path d="M19,15 L21,15" id="路径-4" stroke="#6E6E6E" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"></path>
        </g>
    </g>
</svg>