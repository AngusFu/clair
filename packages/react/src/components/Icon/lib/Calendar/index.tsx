import { IconContainer, IconProps } from "@components/Icon/lib/Container";
import { Calendar } from "@clair/icons";
import { getStyleMergedComponent } from "@src/utils";

export const IconCalendar = getStyleMergedComponent<IconProps>({
  template: Calendar
})(IconContainer);
