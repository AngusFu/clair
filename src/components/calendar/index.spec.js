import { mount } from '@vue/test-utils'
import Calendar from './index.vue'

it('should works in basic mode', () => {
  const wrapper = mount(Calendar, {
    propsData: {
      // specify date, or test could fail
      value: '2017-09-14'
    }
  })
  expect(wrapper).toMatchSnapshot()
})

it('should work with defined props', () => {
  const wrapper = mount(Calendar, {
    propsData: {
      value: '2017-9-15',
      maxDate: '2018-8-15',
      minDate: '2017-8-15',
      pattern: 'yyyy/MM/dd'
    }
  })

  expect(wrapper).toMatchSnapshot()
})

it('should work as month selector', () => {
  const wrapper = mount(Calendar, {
    propsData: {
      type: 'month',
      value: '2017-9-15',
      maxDate: '2018-8-15',
      minDate: '2017-8-15',
      pattern: 'yyyy/MM'
    }
  })

  expect(wrapper).toMatchSnapshot()
})
