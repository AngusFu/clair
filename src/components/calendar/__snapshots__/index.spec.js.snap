// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should work as month selector 1`] = `
<div class="c-calendar md">
  <div class="c-calendar__header">
    <div class="c-calendar__prev-year disabled"><span class="c-icon">​<svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevrons-left" style="vertical-align: text-top;"><polyline points="11 17 6 12 11 7"></polyline><polyline points="18 17 13 12 18 7"></polyline></svg></span></div><a class="c-calendar__prev-month" style="display: none;"><span class="c-icon">​<svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-left" style="vertical-align: text-top;"><polyline points="15 18 9 12 15 6"></polyline></svg></span></a><span class="c-calendar__year">2017</span><span class="c-calendar__spacer" style="display: none;">-</span><span class="c-calendar__month" style="display: none;">09</span><a class="c-calendar__next-month" style="display: none;"><span class="c-icon">​<svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right" style="vertical-align: text-top;"><polyline points="9 18 15 12 9 6"></polyline></svg></span></a><a class="c-calendar__next-year"><span class="c-icon">​<svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevrons-right" style="vertical-align: text-top;"><polyline points="13 17 18 12 13 7"></polyline><polyline points="6 17 11 12 6 7"></polyline></svg></span></a>
  </div>
  <div class="c-calendar__body">
    <table class="c-calendar__month-table">
      <tbody>
        <tr>
          <td><a class="month-cell disabled">一月</a></td>
          <td><a class="month-cell disabled">二月</a></td>
          <td><a class="month-cell disabled">三月</a></td>
        </tr>
        <tr>
          <td><a class="month-cell disabled">四月</a></td>
          <td><a class="month-cell disabled">五月</a></td>
          <td><a class="month-cell disabled">六月</a></td>
        </tr>
        <tr>
          <td><a class="month-cell disabled">七月</a></td>
          <td><a class="month-cell">八月</a></td>
          <td><a class="month-cell active">九月</a></td>
        </tr>
        <tr>
          <td><a class="month-cell">十月</a></td>
          <td><a class="month-cell">十一月</a></td>
          <td><a class="month-cell">十二月</a></td>
        </tr>
      </tbody>
    </table>
    <!---->
  </div>
</div>
`;

exports[`should work with defined props 1`] = `
<div class="c-calendar md">
  <div class="c-calendar__header">
    <div class="c-calendar__prev-year disabled"><span class="c-icon">​<svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevrons-left" style="vertical-align: text-top;"><polyline points="11 17 6 12 11 7"></polyline><polyline points="18 17 13 12 18 7"></polyline></svg></span></div><a class="c-calendar__prev-month"><span class="c-icon">​<svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-left" style="vertical-align: text-top;"><polyline points="15 18 9 12 15 6"></polyline></svg></span></a><span class="c-calendar__year">2017</span><span class="c-calendar__spacer">-</span><span class="c-calendar__month">09</span><a class="c-calendar__next-month"><span class="c-icon">​<svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right" style="vertical-align: text-top;"><polyline points="9 18 15 12 9 6"></polyline></svg></span></a><a class="c-calendar__next-year"><span class="c-icon">​<svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevrons-right" style="vertical-align: text-top;"><polyline points="13 17 18 12 13 7"></polyline><polyline points="6 17 11 12 6 7"></polyline></svg></span></a>
  </div>
  <div class="c-calendar__body">
    <!---->
    <table class="c-calendar__day-table">
      <thead>
        <tr>
          <th>日</th>
          <th>一</th>
          <th>二</th>
          <th>三</th>
          <th>四</th>
          <th>五</th>
          <th>六</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td class="lastmonth"><a data-rowindex="0" data-index="0" class="day-cell">27</a></td>
          <td class="lastmonth"><a data-rowindex="0" data-index="1" class="day-cell">28</a></td>
          <td class="lastmonth"><a data-rowindex="0" data-index="2" class="day-cell">29</a></td>
          <td class="lastmonth"><a data-rowindex="0" data-index="3" class="day-cell">30</a></td>
          <td class="lastmonth"><a data-rowindex="0" data-index="4" class="day-cell">31</a></td>
          <td class="curmonth"><a data-rowindex="0" data-index="5" class="day-cell">1</a></td>
          <td class="curmonth"><a data-rowindex="0" data-index="6" class="day-cell">2</a></td>
        </tr>
        <tr>
          <td class="curmonth"><a data-rowindex="1" data-index="0" class="day-cell">3</a></td>
          <td class="curmonth"><a data-rowindex="1" data-index="1" class="day-cell">4</a></td>
          <td class="curmonth"><a data-rowindex="1" data-index="2" class="day-cell">5</a></td>
          <td class="curmonth"><a data-rowindex="1" data-index="3" class="day-cell">6</a></td>
          <td class="curmonth"><a data-rowindex="1" data-index="4" class="day-cell">7</a></td>
          <td class="curmonth"><a data-rowindex="1" data-index="5" class="day-cell">8</a></td>
          <td class="curmonth"><a data-rowindex="1" data-index="6" class="day-cell">9</a></td>
        </tr>
        <tr>
          <td class="curmonth"><a data-rowindex="2" data-index="0" class="day-cell">10</a></td>
          <td class="curmonth"><a data-rowindex="2" data-index="1" class="day-cell">11</a></td>
          <td class="curmonth"><a data-rowindex="2" data-index="2" class="day-cell">12</a></td>
          <td class="curmonth"><a data-rowindex="2" data-index="3" class="day-cell">13</a></td>
          <td class="curmonth"><a data-rowindex="2" data-index="4" class="day-cell">14</a></td>
          <td class="curmonth active"><a data-rowindex="2" data-index="5" class="day-cell">15</a></td>
          <td class="curmonth"><a data-rowindex="2" data-index="6" class="day-cell">16</a></td>
        </tr>
        <tr>
          <td class="curmonth"><a data-rowindex="3" data-index="0" class="day-cell">17</a></td>
          <td class="curmonth"><a data-rowindex="3" data-index="1" class="day-cell">18</a></td>
          <td class="curmonth"><a data-rowindex="3" data-index="2" class="day-cell">19</a></td>
          <td class="curmonth"><a data-rowindex="3" data-index="3" class="day-cell">20</a></td>
          <td class="curmonth"><a data-rowindex="3" data-index="4" class="day-cell">21</a></td>
          <td class="curmonth"><a data-rowindex="3" data-index="5" class="day-cell">22</a></td>
          <td class="curmonth"><a data-rowindex="3" data-index="6" class="day-cell">23</a></td>
        </tr>
        <tr>
          <td class="curmonth"><a data-rowindex="4" data-index="0" class="day-cell">24</a></td>
          <td class="curmonth"><a data-rowindex="4" data-index="1" class="day-cell">25</a></td>
          <td class="curmonth"><a data-rowindex="4" data-index="2" class="day-cell">26</a></td>
          <td class="curmonth"><a data-rowindex="4" data-index="3" class="day-cell">27</a></td>
          <td class="curmonth"><a data-rowindex="4" data-index="4" class="day-cell">28</a></td>
          <td class="curmonth"><a data-rowindex="4" data-index="5" class="day-cell">29</a></td>
          <td class="curmonth"><a data-rowindex="4" data-index="6" class="day-cell">30</a></td>
        </tr>
        <tr>
          <td class="nextmonth"><a data-rowindex="5" data-index="0" class="day-cell">1</a></td>
          <td class="nextmonth"><a data-rowindex="5" data-index="1" class="day-cell">2</a></td>
          <td class="nextmonth"><a data-rowindex="5" data-index="2" class="day-cell">3</a></td>
          <td class="nextmonth"><a data-rowindex="5" data-index="3" class="day-cell">4</a></td>
          <td class="nextmonth"><a data-rowindex="5" data-index="4" class="day-cell">5</a></td>
          <td class="nextmonth"><a data-rowindex="5" data-index="5" class="day-cell">6</a></td>
          <td class="nextmonth"><a data-rowindex="5" data-index="6" class="day-cell">7</a></td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`should works in basic mode 1`] = `
<div class="c-calendar md">
  <div class="c-calendar__header">
    <div class="c-calendar__prev-year"><span class="c-icon">​<svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevrons-left" style="vertical-align: text-top;"><polyline points="11 17 6 12 11 7"></polyline><polyline points="18 17 13 12 18 7"></polyline></svg></span></div><a class="c-calendar__prev-month"><span class="c-icon">​<svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-left" style="vertical-align: text-top;"><polyline points="15 18 9 12 15 6"></polyline></svg></span></a><span class="c-calendar__year">2018</span><span class="c-calendar__spacer">-</span><span class="c-calendar__month">11</span><a class="c-calendar__next-month"><span class="c-icon">​<svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right" style="vertical-align: text-top;"><polyline points="9 18 15 12 9 6"></polyline></svg></span></a><a class="c-calendar__next-year"><span class="c-icon">​<svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevrons-right" style="vertical-align: text-top;"><polyline points="13 17 18 12 13 7"></polyline><polyline points="6 17 11 12 6 7"></polyline></svg></span></a>
  </div>
  <div class="c-calendar__body">
    <!---->
    <table class="c-calendar__day-table">
      <thead>
        <tr>
          <th>日</th>
          <th>一</th>
          <th>二</th>
          <th>三</th>
          <th>四</th>
          <th>五</th>
          <th>六</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td class="lastmonth"><a data-rowindex="0" data-index="0" class="day-cell">28</a></td>
          <td class="lastmonth"><a data-rowindex="0" data-index="1" class="day-cell">29</a></td>
          <td class="lastmonth"><a data-rowindex="0" data-index="2" class="day-cell">30</a></td>
          <td class="lastmonth"><a data-rowindex="0" data-index="3" class="day-cell">31</a></td>
          <td class="curmonth"><a data-rowindex="0" data-index="4" class="day-cell">1</a></td>
          <td class="curmonth"><a data-rowindex="0" data-index="5" class="day-cell">2</a></td>
          <td class="curmonth"><a data-rowindex="0" data-index="6" class="day-cell">3</a></td>
        </tr>
        <tr>
          <td class="curmonth"><a data-rowindex="1" data-index="0" class="day-cell">4</a></td>
          <td class="curmonth"><a data-rowindex="1" data-index="1" class="day-cell">5</a></td>
          <td class="curmonth"><a data-rowindex="1" data-index="2" class="day-cell">6</a></td>
          <td class="curmonth"><a data-rowindex="1" data-index="3" class="day-cell">7</a></td>
          <td class="curmonth"><a data-rowindex="1" data-index="4" class="day-cell">8</a></td>
          <td class="curmonth"><a data-rowindex="1" data-index="5" class="day-cell">9</a></td>
          <td class="curmonth"><a data-rowindex="1" data-index="6" class="day-cell">10</a></td>
        </tr>
        <tr>
          <td class="curmonth"><a data-rowindex="2" data-index="0" class="day-cell">11</a></td>
          <td class="curmonth active"><a data-rowindex="2" data-index="1" class="day-cell">12</a></td>
          <td class="curmonth"><a data-rowindex="2" data-index="2" class="day-cell">13</a></td>
          <td class="curmonth"><a data-rowindex="2" data-index="3" class="day-cell">14</a></td>
          <td class="curmonth"><a data-rowindex="2" data-index="4" class="day-cell">15</a></td>
          <td class="curmonth"><a data-rowindex="2" data-index="5" class="day-cell">16</a></td>
          <td class="curmonth"><a data-rowindex="2" data-index="6" class="day-cell">17</a></td>
        </tr>
        <tr>
          <td class="curmonth"><a data-rowindex="3" data-index="0" class="day-cell">18</a></td>
          <td class="curmonth"><a data-rowindex="3" data-index="1" class="day-cell">19</a></td>
          <td class="curmonth"><a data-rowindex="3" data-index="2" class="day-cell">20</a></td>
          <td class="curmonth"><a data-rowindex="3" data-index="3" class="day-cell">21</a></td>
          <td class="curmonth"><a data-rowindex="3" data-index="4" class="day-cell">22</a></td>
          <td class="curmonth"><a data-rowindex="3" data-index="5" class="day-cell">23</a></td>
          <td class="curmonth"><a data-rowindex="3" data-index="6" class="day-cell">24</a></td>
        </tr>
        <tr>
          <td class="curmonth"><a data-rowindex="4" data-index="0" class="day-cell">25</a></td>
          <td class="curmonth"><a data-rowindex="4" data-index="1" class="day-cell">26</a></td>
          <td class="curmonth"><a data-rowindex="4" data-index="2" class="day-cell">27</a></td>
          <td class="curmonth"><a data-rowindex="4" data-index="3" class="day-cell">28</a></td>
          <td class="curmonth"><a data-rowindex="4" data-index="4" class="day-cell">29</a></td>
          <td class="curmonth"><a data-rowindex="4" data-index="5" class="day-cell">30</a></td>
          <td class="nextmonth"><a data-rowindex="4" data-index="6" class="day-cell">1</a></td>
        </tr>
        <tr>
          <td class="nextmonth"><a data-rowindex="5" data-index="0" class="day-cell">2</a></td>
          <td class="nextmonth"><a data-rowindex="5" data-index="1" class="day-cell">3</a></td>
          <td class="nextmonth"><a data-rowindex="5" data-index="2" class="day-cell">4</a></td>
          <td class="nextmonth"><a data-rowindex="5" data-index="3" class="day-cell">5</a></td>
          <td class="nextmonth"><a data-rowindex="5" data-index="4" class="day-cell">6</a></td>
          <td class="nextmonth"><a data-rowindex="5" data-index="5" class="day-cell">7</a></td>
          <td class="nextmonth"><a data-rowindex="5" data-index="6" class="day-cell">8</a></td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
